components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -626,25 +650,143 @@ void SnsResolverTask::OnFetchSolRecordRegistryState(
+    SetAddressResult(domain_name_registry_state_->owner);
+  }
+}
+
+void SnsResolverTask::FetchUrlRecordRegistryState() {
+  auto url_record_address = GetDomainKey("url." + domain_, true);
+  if (!url_record_address) {
+    SetError(MakeInternalError());
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&SnsResolverTask::OnFetchUrlRecordRegistryState,
+                     weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(solana::getAccountInfo(url_record_address->ToBase58()),
+                  std::move(internal_callback),
+                  solana::ConverterForGetAccountInfo());
+}
+
+void SnsResolverTask::OnFetchUrlRecordRegistryState(
+    APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(api_request_result.body(), &account_info)) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  if (!account_info) {
+    // No url record account, will proceed with ipfs record.
+    url_record_check_done_ = true;
+    return;
+  }
+
+  auto url_record_name_registry_state =
+      FromBase64<NameRegistryState>(account_info->data);
+  if (!url_record_name_registry_state) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  // Trim possible zeros.
+  std::string url_string(
+      std::string(url_record_name_registry_state->data.begin(),
+                  url_record_name_registry_state->data.end())
+          .c_str());
+
+  GURL url(url_string);
+  if (!url.is_valid()) {
+    SetError(MakeInvalidParamsError());
+    return;
+  }
+
+  SetUrlResult(std::move(url));
+}
+
+void SnsResolverTask::FetchIpfsRecordRegistryState() {
+  auto ipfs_record_address = GetDomainKey("IPFS." + domain_, true);
+  if (!ipfs_record_address) {
+    SetError(MakeInternalError());
+    ScheduleWorkOnTask();
+    return;
   }
+
+  auto internal_callback =
+      base::BindOnce(&SnsResolverTask::OnFetchIpfsRecordRegistryState,
+                     weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(solana::getAccountInfo(ipfs_record_address->ToBase58()),
+                  std::move(internal_callback),
+                  solana::ConverterForGetAccountInfo());
+}
+
+void SnsResolverTask::OnFetchIpfsRecordRegistryState(
+    APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(api_request_result.body(), &account_info)) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  if (!account_info) {
+    // No ipfs record account, resolve as error.
+    SetError(MakeInvalidParamsError());
+    return;
+  }
+
+  auto ipfs_record_name_registry_state =
```

Comment:
A reference to how IPFS record looks like?

Reply:
https://github.com/brave/brave-core/pull/15863/commits/cd6f31ee6d23c7c35177374065e5a69d9ddf6042